### PR TITLE
Add dockerfile to build docker image

### PR DESCRIPTION
**Pull request by lvyelin**
## Issue

Add dockerfile to build docker image

## Test

Docker image harmonyone/node:master tested.

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

File: scripts/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM alpine
+
+RUN apk add --no-cache bash libstdc++ gmp-dev libc6-compat && ln -s libcrypto.so.1.1 /lib/libcrypto.so.10
+
+EXPOSE 9000/tcp 14555/tcp 6000/tcp
```
Comment (LeoHChen) [inline scripts/docker/Dockerfile]: can we add some comment here about the port? 

9000 is the base port of harmony node
14555 is the RPC server port
6000 is the REST server port?

File: scripts/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+VOLUME ["/harmony/db", "/harmony/log"]
+
+ENV BN_MA "/ip4/100.26.90.187/tcp/9874/p2p/Qmdfjtk6hPoyrH1zVD9PEH4zfWLo38dP2mDvvKXfh3tnEv,/ip4/54.213.43.194/tcp/9874/p2p/QmZJJx6AdaoEkGLrYG4JeLCKeCKDjnFz2wfHNHxAqFSGA9"
```
Comment (LeoHChen) [inline scripts/docker/Dockerfile]: Also, comment here please.  This list of BN is for cello release.

File: scripts/docker/docker-node.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+DOCKER_IMAGE=harmonyone/node:master
+
+if [ -z "$(which docker)" ]; then
+  echo "docker is not installed."
+  echo "Please chec https://docs.docker.com/install/ to get docker installed"
```
Comment (LeoHChen) [inline scripts/docker/docker-node.sh]: typo: chec => check

File: scripts/docker/docker-node.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+fi
+
+port_base=
+
+while getopts "p:" opt; do
+  case "$opt" in
+    p) port_base="$OPTARG";;
+    *) exit 1;;
```
Comment (LeoHChen) [inline scripts/docker/docker-node.sh]: print usage here?

File: scripts/docker/docker-node.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  echo "port base cannot be less than 4000"
+  exit 1
+fi
+
+if [ "$port_base" -gt 59900 ]; then
+  echo "port base cannot be greater than 59900"
+  exit 1
+fi
+
+port_1=$(( $port_base - 3000 ))
```
Comment (LeoHChen) [inline scripts/docker/docker-node.sh]: nit: port_1 => rest_port
port_2 => rpc_port

File: scripts/docker/run [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    ip=($ip)
+    IFS=$OIFS
+    [[ ${ip[0]} -le 255 && ${ip[1]} -le 255 \
+      && ${ip[2]} -le 255 && ${ip[3]} -le 255 ]]
+    stat=$?
+  fi
+  return $stat
+}
+
+PUB_IP=$(wget -qO- https://diagnostic.opendns.com/myip)
```
Comment (LeoHChen) [inline scripts/docker/run]: Does it return ipv4 address only?  Check my PR. https://github.com/harmony-one/harmony/pull/715

**Comment by LeoHChen** [comment]
Jenkins, test this please.

**Pull request merged**